server/src/main/java/org/elasticsearch/index/mapper/IdFieldMapper.java

```diff
@@ -87,7 +87,9 @@
         }
     }
 
-    public static final TypeParser PARSER = new FixedTypeParser(c -> new IdFieldMapper(c.isIdFieldDataEnabled()));
+    public static final IdFieldMapper DISABLED = new IdFieldMapper(() -> false);
```

Comment:
I think we can take this further, let's have two static final values here, `FIELD_DATA_ENABLED` and `FIELD_DATA_DISABLED`, with a static selector method that returns either one of them, and keep the constructor private so that we only ever use these two instances.

Reply:
But the setting can be updated dynamically? How would we switch out all the references to the mapper that might be flying around the node on setting update

Reply:
Ah, you're quite right, it's an updateable setting.  Can we still rename `DISABLED` no `NO_FIELD_DATA` or something similar to make it clear that we're not disabling the whole mapper?

Reply:
++ makes sense, renamed :)